src/app/data-model/DecodableList.h

```diff
@@ -148,13 +173,18 @@ class DecodableList
 template <typename X>
 CHIP_ERROR Decode(TLV::TLVReader & reader, DecodableList<X> & x)
 {
-    VerifyOrReturnError(reader.GetType() == TLV::kTLVType_Array, CHIP_ERROR_SCHEMA_MISMATCH);
-
-    TLV::TLVType type;
-
-    ReturnErrorOnFailure(reader.EnterContainer(type));
-    x.SetReader(reader);
-    ReturnErrorOnFailure(reader.ExitContainer(type));
+    if (reader.GetType() == TLV::kTLVType_Null)
```

Comment:
What the spec says is:

1) Null and empty list must have the same semantics, _for a list that is nullable_.
2) Null is encoded as TLV null.
3) Empty list is encoded as empty TLV array.
4) Some lists are nullable, some are not.

There was some discussion about alternate encodings for empty non-nullable lists, but that's not where the spec landed in the end.

Nullable lists are represented in our C++ as `Nullable<DecodableList>` and you can decode into that and get the handling of TLV null.

Non-nullable lists are represented as just `DecodableList`, and a TLV null on the wire is not a valid encoding for those.  This change seems to want to allow TLV null on the wire for non-nullable lists...

Why are we making this change?

Reply:
Ah, good point.

Reply:
Decode changes removed. Other changes kept with adjusted documentation.